### ISSUE TITLE
YARP and NotFoundHandler pipeline processing order

### ISSUE THREAD

**Issue by AgentLintZeal** [microsoft/reverse-proxy#2101]
I have an installation using both the NotFoundHandler and Yarp reverse proxy. The NotFoundHandler is being loaded in the ConfigureServices method and Yarp is loaded after in the Application builder Configure method.

Everything works as expected until you try to redirect a URL that also needs to be reverse proxied. In this instance, the reverse proxy gets the URL before the NotFoundHandler and this results in a 404.  The redirector should change the path and then let the reverse proxy alter the URL, but the NotFoundHandler never gets called.

For example:

This is the NotFoundHandler URL that is mapped
- https://website.domain.com/cars/4WD/explorer to https://website.domain.com/cars/ford/explorer

Then the reverse proxy needs to adjust the browser URL to look like this:
-  https://vehicles.domain.com/ford/explorer

The reverse proxy sees:
- https://website.domain.com/cars/4WD/explorer 
and changes it to:
- https://vehicles.domain.com/4WD/explorer

before the redirector can change the URL - even though it should be coming first in the pipeline.

Yarp.ReverseProxy version 2.0.0
Geta.NotFoundHandler.Optimiziely version 5.0.4
EPiServer.CMS version 12.13.2
Windows

**Comment by karelz** [comment]
Triage: Looks like IIS routing problem to us, involving 3rd party (Geta) Nuget package.
We do not have expertise in IIS routing. Can you either debug it, or ask on IIS forums?

**Comment by AgentLintZeal** [comment]
I took a while to respond as I have been working the issue. It seems that Yarp an the Geta redirector function at very different levels so I ended up adding a piece of code to read the Geta DB table and perform the redirect as part of a custom proxy transformer.  This works for 90% of the situations however I was not able to find a way to cancel or abort a proxy that has already begun. This is needed due to CloudFlare complaining if the redirection ends up not being needed. It is hard to explain without a lot of supporting documentation to recreate.

The short version is: How can I cancel a proxy transformation that is in progress and have the normal routing continue. It seems that once a proxy request has started, it has to complete. Attempts to cancel it just result in http errors or a blank page being rendered.
